src/en/kami18/src/eu/kanade/tachiyomi/extension/en/kami18/Kami18.kt

```diff
@@ -0,0 +1,178 @@
+package eu.kanade.tachiyomi.extension.en.kami18
+
+import eu.kanade.tachiyomi.network.GET
+import eu.kanade.tachiyomi.network.asObservableSuccess
+import eu.kanade.tachiyomi.source.model.FilterList
+import eu.kanade.tachiyomi.source.model.MangasPage
+import eu.kanade.tachiyomi.source.model.Page
+import eu.kanade.tachiyomi.source.model.SChapter
+import eu.kanade.tachiyomi.source.model.SManga
+import eu.kanade.tachiyomi.source.online.HttpSource
+import eu.kanade.tachiyomi.util.asJsoup
+import okhttp3.Headers
+import okhttp3.HttpUrl.Companion.toHttpUrl
+import okhttp3.Request
+import okhttp3.Response
+import org.jsoup.nodes.Element
+import rx.Observable
+import java.lang.Exception
+import java.text.SimpleDateFormat
+import java.util.Locale
+
+class Kami18() : HttpSource() {
+
+    override val name = "18Kami"
+
+    override val lang = "en"
+
+    override val baseUrl = "https://18kami.com"
+
+    private val baseImageUrl = "$baseUrl/media/photos"
+
+    override val supportsLatest = true
+
+    override val client = network.cloudflareClient
+
+    override fun headersBuilder() = Headers.Builder().apply {
+        add("Referer", "$baseUrl/")
+    }
+
+    override fun popularMangaRequest(page: Int): Request {
+        return GET("$baseUrl/albums?o=mv&page=$page", headers)
+    }
+
+    override fun popularMangaParse(response: Response): MangasPage {
+        val document = response.asJsoup()
+
+        val entries = document.select(".image-container")
+        val hasNextPage = document.selectFirst(".prevnext") != null
+
+        return MangasPage(entries.map(::popularMangaFromElement), hasNextPage)
+    }
+
+    private fun popularMangaFromElement(element: Element) = SManga.create().apply {
+        setUrlWithoutDomain(element.selectFirst("a[href*=album]")!!.absUrl("href"))
+        title = element.selectFirst("img")!!.attr("title")
+        thumbnail_url = element.selectFirst("img")?.let { img ->
+            img.absUrl("src").takeIf { !it.contains("blank") } ?: img.absUrl("data-original")
+        }
+    }
+
+    override fun latestUpdatesRequest(page: Int): Request {
+        return GET("$baseUrl/albums?o=mr&page=$page", headers)
+    }
+
+    override fun latestUpdatesParse(response: Response) = popularMangaParse(response)
+
+    override fun searchMangaRequest(page: Int, query: String, filters: FilterList): Request {
+        if (query.isNotEmpty()) {
+            val url = baseUrl.toHttpUrl().newBuilder().apply {
+                addPathSegment("search")
+                addPathSegment("photos")

```

Comment:
```suggestion
            val url = "$baseUrl/search/photos".toHttpUrl().newBuilder().apply {
```

Not sure if `addPathSegment` adds a trailing slash, but since it's static I don't see an issue with this approach either way